src/ucp/rndv/rndv_ppln.c

```diff
@@ -85,52 +97,76 @@ ucp_proto_rndv_ppln_probe(const ucp_proto_init_params_t *init_params)
         return;
     }
 
+    ack_params.max_length = SIZE_MAX;
     /* Add each proto as a separate variant */
     ucs_array_for_each(proto, &select_elem->proto_init.protocols) {
-        if (ucp_proto_id_field(proto->proto_id, flags) &
-            UCP_PROTO_FLAG_INVALID) {
+        proto_flags = ucp_proto_id_field(proto->proto_id, flags);
+        if (proto_flags & UCP_PROTO_FLAG_INVALID) {
             continue;
         }
 
-        ucs_assert(proto->caps.num_ranges > 0);
+        ucs_assert(!ucp_proto_perf_is_empty(proto->perf));
 
-        frag_range = proto->caps.ranges + (proto->caps.num_ranges - 1);
-        if (frag_range->max_length == 0) {
+        /* Add the single range of the pipeline protocol to ppln_caps */
+        status = ucp_proto_perf_create("pipeline", &ppln_perf);
+        if (status != UCS_OK) {
             continue;
         }
 
+        first_seg = ucp_proto_perf_find_segment_lb(proto->perf, 0);
```

Comment:
initialize first_seg right before line 123 - where it is used

Reply:
Done